<!-- 
	PLAN_RBP_RANDOM_FOREST.LAUNCH
	Multi-agent Trajectory Planning and Simulation using Relative Bernstein Polynomial
        
	USAGE : At terminal, roslaunch swarm_planner plan_rbp_random_forest.launch
        
        TIPS : 1. You may turn off 'runsim', 'log' arguments to check the correct computation time.
               2. If you want to retry this at the same world, 
                  (1) Do not terminate the previous one, and launch follows at the other terminal. 
                      roslaunch swarm_planner map_saver.launch
                  (2) Terminate the previous one, and turn on 'replay' argument in this launch file. 
                  (3) Relaunch it.
        
	Jungwon Park
	Lab for Autonomous Robotics Research
	Seoul National Univ. 
	wjstk335@gmail.com
-->

<launch>
<!-- Arguments Start -->

  <!-- Simulation -->
  <arg name="mission"               default="mission_32agents_15.json"/>
  <arg name="runsim"                default="false"/>
  <arg name="log"                   default="false"/>
  
  <!-- Environment -->
  <arg name="world_x_min"           default="-5"/>
  <arg name="world_y_min"           default="-5"/>
  <arg name="world_z_min"           default="0"/>
  <arg name="world_x_max"           default="5"/>
  <arg name="world_y_max"           default="5"/>
  <arg name="world_z_max"           default="2.5"/>
  <arg name="world_resolution"      default="0.1"/>

  <!-- Obstacles -->
  <arg name="world_obs_num"         default="20"/>
  <arg name="world_r_min"           default="0.3"/>
  <arg name="world_r_max"           default="0.3"/>
  <arg name="world_h_min"           default="1.0"/>
  <arg name="world_h_max"           default="2.5"/>
  <arg name="world_margin"          default="0.5"/>
  
  <!-- InitTrajPlanner Parameters -->
  <arg name="InitTraj_planner_type" default="0"/>    <!-- 0: ECBS, 1: SIPP -->
  <arg name="ecbs_w"                default="1.3"/>  <!-- ECBS only -->
  <arg name="grid_xy_res"           default="0.5"/>
  <arg name="grid_z_res"            default="1.0"/>
  <arg name="grid_margin"           default="0.2"/>

  <!-- BoxGenerator Parameters-->
  <arg name="box_xy_res"            default="0.1"/>
  <arg name="box_z_res"             default="0.1"/>

  <!-- SwarmPlanner Parameters -->
  <arg name="plan_time_step"        default="1"/>
  <arg name="plan_downwash"         default="2.0"/>
  <arg name="plan_sequential"       default="true"/>
  <arg name="plan_iteration"        default="1"/>    <!-- sequential only -->
  <arg name="plan_batch"            default="2"/>
  <arg name="plan_gn"               default="0"/>
  <arg name="plan_n"                default="5"/>
  <arg name="plan_phi"              default="3"/>  

<!-- Arguments End -->

<!-- Nodes Start -->
  <node pkg="swarm_planner" type="swarm_traj_planner_rbp_test_all" name="swarm_traj_planner_rbp_test_all" output="screen">
    <param name="mission"                    value="$(find swarm_planner)/missions/$(arg mission)" />
    <param name="log"                        value="$(arg log)" />
        
    <param name="world/x_min"                value="$(arg world_x_min)" />
    <param name="world/y_min"                value="$(arg world_y_min)" />
    <param name="world/z_min"                value="$(arg world_z_min)" />
    <param name="world/x_max"                value="$(arg world_x_max)" />
    <param name="world/y_max"                value="$(arg world_y_max)" />
    <param name="world/z_max"                value="$(arg world_z_max)" />

    <param name="ecbs/w"                     value="$(arg ecbs_w)" />
    <param name="grid/xy_res"                value="$(arg grid_xy_res)" />
    <param name="grid/z_res"                 value="$(arg grid_z_res)" />
    <param name="grid/margin"                value="$(arg grid_margin)" />

    <param name="box/xy_res"                 value="$(arg box_xy_res)" />
    <param name="box/z_res"                  value="$(arg box_z_res)" />

    <param name="plan/initTraj_planner_type" value="$(arg InitTraj_planner_type)" />
    <param name="plan/time_step"             value="$(arg plan_time_step)" />
    <param name="plan/downwash"              value="$(arg plan_downwash)" />
    <param name="plan/n"                     value="$(arg plan_n)" />
    <param name="plan/phi"                   value="$(arg plan_phi)" />

    <param name="plan/sequential"            value="$(arg plan_sequential)" />
    <param name="plan/group"                 value="$(arg plan_batch)" />
    <param name="plan/gn"                    value="$(arg plan_gn)" />
  </node>
<!-- Nodes End -->
</launch>
